sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/http/client/DefaultHttpClientProvider.java

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.implementation.http.client;
+
+import com.generic.core.http.client.DefaultHttpClientBuilder;
+import com.generic.core.http.client.HttpClient;
+import com.generic.core.http.client.HttpClientProvider;
+
+/**
+ * An {@link HttpClientProvider} that provides an implementation of HttpClient based on native JDK HttpURLConnection.
+ */
+public class DefaultHttpClientProvider extends HttpClientProvider {
```

Comment:
Do we actually need this? Should the default HttpClient just be used if nothing is found on the classpath?

Reply:
The default `HttpClient` is used when no provider is found on the classpath.

Reply:
Alan's question is more - why do we need this class at all. The default HTTP Client is our fallback client, we don't also need it to be an SPI implementation.

Reply:
Yep, it's also a bit of an odd pattern for the library defining the service provider interface to also have an implementation of it as that means you'll always have something on the classpath (or at least should). And that sort of defeats the purpose of the SPI.